internal/resources/cluster/data_source_cluster.go

```diff
@@ -45,7 +45,7 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 		if err != nil {
 			if clienterrors.IsNotFoundError(err) {
 				_ = schema.RemoveFromState(d, m)
-				return false, nil
+				return false, errors.Wrapf(err, "Tanzu Mission Control cluster entry not found, name : %s", d.Get(NameKey))
```

Comment:
we should not return this error i guess

Reply:
The plugin crashes otherwise, at this code point the error should be returned in case of cluster not found. 
If we added the condition of early return without the error, then only terraform emitted warnings, that to which are only catched if running provider in DEBUG mode. 

The warnings- 

`2023-03-09T13:15:55.888+0530 [WARN]  Provider "provider[\"vmware/dev/tanzu-mission-control\"]" produced an unexpected new value for data.tanzu-mission-control_cluster.tf-cluster.
      - .management_cluster_name: was cty.StringVal("vasundharas-tkgs"), but now null
      - .name: was cty.StringVal("tf-cluster-new-2"), but now null
      - .provisioner_name: was cty.StringVal("testns"), but now null
`

Reply:
and this data source handler function is also invoked by resource, so check for the possible corner cases